packages/react-router/src/useBlocker.tsx

```diff
@@ -3,25 +3,70 @@ import { useRouter } from './useRouter'
 import type { BlockerFn } from '@tanstack/history'
 import type { ReactNode } from './route'
 
-export function useBlocker(
-  blockerFn: BlockerFn,
-  condition: boolean | any = true,
-): void {
+type BlockerResolver = {
+  status: 'idle' | 'blocked'
+  proceed: () => void
+  reset: () => void
+}
+
+export function useBlocker(opts?: {
```

Comment:
Switching to object syntax, makes this a breaking change. I think we may have to stick with the existing API surfaces.

Reply:
We follow semver and this'd be a breaking change to the user.

Considering we don't want to introduce compexity with function overloads, I think we'd have to go with the alternatives here (like `useBlocker(() => void, value)`), or simply export a new useBlocker hook altogether (with a new name) and preserve the old hook as-is.

@schiller-manuel thoughts on this?

Reply:
we must avoid breaking changes.
instead, I suggest supporting both signatures like this (pseudocode):

```tsx
function useBlocker(
blockerFnOrOpts: BlockerFn | BlockerOpts,
condition?: boolean | any)
{
if (typeof blockerFnOrOpts === 'function') {
// function was passed
}
else {
// opts were passed
}
}
```

then deprecate the old syntax and get rid of it with v2

Reply:
I updated the API, I don't know why now tests are failing on the start package though... on local tsc doesn't complain

Reply:
Rebasing with main should fix this now.